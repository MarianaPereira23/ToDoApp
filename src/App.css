@import url('https://fonts.googleapis.com/css2?family=Open+Sans+Condensed:wght@300&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Open Sans Condensed', sans-serif;
}

html {
  width: 100vw;
  height: 100%;
}

body {
  width: 100vw;
  min-height: 100vh;
  background: url('./images/home.jpeg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  position: fixed;
}

.App {
  height: 100vh;
  margin: 0 auto;
  overflow-y: auto;
}

.page-container {
  max-width: 1440px;
  min-height: 85vh;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.5rem 0;
}
