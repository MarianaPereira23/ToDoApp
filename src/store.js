import { createStore } from 'redux';
import { loadState, saveState } from './localStorage';
import todoReducer from './reducers/reducers';

const savedState = loadState();

const initialState = savedState;

const store = createStore(todoReducer, initialState);

store.subscribe(() => saveState(store.getState()));

export default store;
