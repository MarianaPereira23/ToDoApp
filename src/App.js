/* eslint-disable react/jsx-first-prop-new-line */
/* eslint-disable react/jsx-max-props-per-line */
/* eslint-disable no-param-reassign */
import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Header from './components/Header/Header';
import Form from './components/Form/Form';
import EditForm from './components/EditForm/EditForm';
import TodoList from './components/TodoList/TodoList';
import ToggleDisplay from './components/ToggleDisplay/ToggleDisplay';
import './App.css';

const App = () => (
  <div className="App">
    <Header />
    <Routes>
      <Route path="/" element={(
        <section className="page-container">
          <Form />
          <ToggleDisplay />
          <TodoList />
        </section>
      )} />
      <Route path="/edit/:id" element={(
        <section className="page-container">
          <EditForm />
        </section>
      )} />
    </Routes>
  </div>
);

export default App;
