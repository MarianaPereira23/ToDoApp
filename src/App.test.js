/* eslint-disable max-len */
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from './store';
import App from './App';

describe('Redux store', () => {
  it('gets todos', () => {
    const todoList = store.getState();
    expect(todoList).toEqual([]);
  });
});
describe('App', () => {
  const div = document.createElement('div');
  test('Renders without crashing', () => {
    ReactDOM.render(<BrowserRouter><Provider store={store}><App /></Provider></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test('App snapshot', () => {
    const AppComponent = renderer.create(<BrowserRouter><Provider store={store}><App /></Provider></BrowserRouter>).toJSON();
    expect(AppComponent).toMatchSnapshot();
  });
  test('Click events work - add task, change task to done and remove task', () => {
    render(<BrowserRouter><Provider store={store}><App /></Provider></BrowserRouter>);
    const titleInput = screen.getByPlaceholderText('Task Title*');
    fireEvent.change(titleInput, { target: { value: 'Do stuff' } });
    expect(titleInput.value).toEqual('Do stuff');

    fireEvent.click(screen.getAllByRole('button')[0]);
    expect(screen.getByText('Do stuff')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Edit'));

    const titleChangeInput = screen.getByDisplayValue('Do stuff');
    fireEvent.change(titleChangeInput, { target: { value: 'Do a lot of stuff' } });
    expect(titleChangeInput.value).toEqual('Do a lot of stuff');

    fireEvent.click(screen.getAllByRole('button')[0]);
    expect(screen.getByText('Do a lot of stuff')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Do a lot of stuff'));
    expect(screen.getByText('Do a lot of stuff').parentNode.classList.length).toBe(2);
    expect(screen.getByText('Do a lot of stuff').parentNode.classList[1]).toBe('done');

    fireEvent.click(screen.getByText('Remove'));

    expect(screen.queryByText('Do a lot of stuff')).toBeNull();
  });
});
