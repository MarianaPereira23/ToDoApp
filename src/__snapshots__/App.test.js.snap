// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot 1`] = `
<div
  className="App"
>
  <header
    className="page-header"
  >
    <h1
      className="page-title"
    >
      Todays Goals
    </h1>
  </header>
  <section
    className="page-container"
  >
    <section
      className="form-container"
    >
      <form
        className="addTodo-form"
        onSubmit={[Function]}
      >
        <button
          className="addTodo-form__add-btn"
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus fa-w-14 add-button__icon"
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
        <input
          className="addTodo-form__title"
          placeholder="Task Title*"
          required={true}
          type="text"
        />
        <input
          className="addTodo-form__description"
          placeholder="Description"
          type="textarea"
        />
      </form>
    </section>
    <div
      className="toggle-container hidden"
    >
      <button
        className="toggle-container__button"
        onClick={[Function]}
        type="submit"
      >
        Hide done tasks
      </button>
    </div>
    <section
      className="todo-container"
    />
  </section>
</div>
`;
